core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -602,7 +602,8 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime, threadNamePr
                                       maxMessageSize = config.messageMaxBytes,
                                       maxIoBytesPerSecond = config.logCleanerIoMaxBytesPerSecond,
                                       backOffMs = config.logCleanerBackoffMs,
-                                      enableCleaner = config.logCleanerEnable)
+                                      enableCleaner = config.logCleanerEnable,
+                                      brokerCompressionType = config.compressionType)
```

Comment:
Also, unrelated to this patch, but can you please move the following comment to before we call `fetchAllTopicConfigs` instead of after?

```scala
// read the log configurations from zookeeper
```